fty-shm-cleanup/src/fty_shm_cleanup.cc

```diff
@@ -50,108 +48,126 @@ static int parse_ttl(char* ttl_str, time_t& ttl)
-      std::string filename(directory_path);
-      filename.append("/").append(ent->d_name);
-      if(ent->d_type == DT_DIR) {
-        fty_shm_cleanup(filename, false);
-      } else {
-        clean_outdated_data(filename);
-      }
+// cleanup outdated metrics from PATH
+// returns 0 if success, else <0
+static int fty_shm_cleanup(const std::string& directory_path, size_t &removedFilesCnt, bool verbose)
+{
+    if (verbose) {
+        log_info("shm cleanup directory '%s'", directory_path.c_str());
     }
-    closedir (dir);
-  }
-  return 0;
-}
 
-static const char help_text[]
-    = "fty-shm-cleanup [options] ...\n"
-      "  -v, --verbose         show verbose output\n"
-      "  -h, --help            display this help text and exit\n";
+    DIR *dir = opendir(directory_path.c_str());
+    if (dir == nullptr) {
+        log_error("opendir %s failed (%s)", directory_path.c_str(), strerror(errno));
+        return -1;
+    }
+
+    struct dirent *ent;
+    while ((ent = readdir(dir)) != nullptr) {
+        if (strcmp(ent->d_name, ".") == 0 || strcmp(ent->d_name, "..") == 0)
+            continue;
+        std::string filename(directory_path);
+        filename.append("/").append(ent->d_name);
+        if (ent->d_type == DT_DIR) { // recursive
+            fty_shm_cleanup(filename, removedFilesCnt, verbose);
+        }
+        else {
+            if (clean_outdated_data(filename) == 0) {
+                removedFilesCnt++;
+            }
+        }
+    }
+
+    closedir(dir);
+    return 0;
+}
 
 int main(int argc, char* argv[])
 {
+    const char* agent_name = "fty-shm-cleanup";
+    const std::string path{"/run/42shm"};
     bool verbose = false;
 
-    static struct option long_opts[] = {
-        { "help", no_argument, 0, 'h' },
-        { "verbose", no_argument, 0, 'v' }
-    };
-
-    std::string path("/run/42shm");
-
-    int c = 0;
-    while (c >= 0) {
-        c = getopt_long(argc, argv, "hv", long_opts, 0);
-
-        switch (c) {
-        case 'v':
-            verbose = true;
-            break;
-        case 'h':
-            std::cout << help_text;
-            return 0;
-        case '?':
-            std::cerr << help_text;
-            return 1;
-        default:
-            // Should not happen
-            c = -1;
+    // handle args
+    {
+        static const char help_text[]
+            = "fty-shm-cleanup [options] ...\n"
+              "  -v    verbose output\n"
+              "  -h    display this help text and exit\n";
+
+        int argn;
+        for (argn = 1; argn < argc; argn++) {
+            const char* arg = argv[argn];
+            if (strcmp(arg, "-v") == 0) {
+                verbose = true;
+            }
+            else if (strcmp(arg, "-h") == 0) {
+                std::cout << help_text;
+                return EXIT_SUCCESS;
+            }
+            else {
+                std::cerr << help_text;
+                std::cerr << "unknown argument '" << std::string(arg) << "'" << std::endl;
```

Comment:
you can create this string on line 146 and manipulate by '==' instead of 'strcmp()'